### DISCUSSION TITLE
Creating a target when several files are created at the same time

### DISCUSSION THREAD

**Discussion by gueyenono** [ropensci/targets#415] [General]
I would like to know how to specify the target when several files are written to disk with, say, the `purrr::walk()` function. In the single file case, it is fairly simple and the book explains it very well. What I need is for my function to the path of the created file:

```
library(targets)
library(ggplot2)

list(
  tar_target(
    my_plot, {
      gg <- ggplot(mtcars, aes(x = wt, y = mpg)) + geom_col()
      ggsave(gg, "my_plot.png")
      "my_plot.png"
    },
    format = "file"
  )
)
```

However, when I have a case like what follows, I am not sure how to go about it. So, here, I have a list of `ggplot` objects and I am saving them all with the `purrr::walk` function:

```
library(targets)
library(ggplot2)

list(
  
  tar_target(
    my_plots, {
      gg <- ggplot(mtcars, aes(x = wt, y = mpg))
      list(
        gg1 = gg + geom_col(),
        gg2 = gg + geom_line()
      )
    }
  ),
  
  tar_target(
    save_my_plots, {
      plot_names <- paste0("gg", 1:2, ".png")
      purrr::walk2(my_plots, function(gg, name){
        ggsave(gg, name)
      })
    },
    format = "file"
  )
)
```

Will adding a file path at the end of `purrr::walk2()` work?

Thank you

**Comment by wlandau** [comment]
You have the right idea. If the `save_my_plots` target returns `plot_names`, that file target will work. A file target can track multiple files as long as those paths are listed in the returned character vector.

**Reply by gueyenono** [reply]
Okay, well noted @wlandau. Thank you very much.
